service/src/main/java/org/broadinstitute/listener/relay/inspectors/HeaderLoggerInspector.java

```diff
@@ -20,27 +24,69 @@ public class HeaderLoggerInspector implements RequestInspector {
   @Override
   public boolean inspectWebSocketUpgradeRequest(
       @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
-
-    logger.debug("WebSocket Upgrade Request: {}", relayedHttpListenerRequest.getUri());
-
-    logHeaders(relayedHttpListenerRequest.getHeaders());
-
+    logRequest(relayedHttpListenerRequest, OffsetDateTime.now(), "WEBSOCKET_UPGRADE_REQUEST");
     return true;
   }
 
   @Override
   public boolean inspectRelayedHttpRequest(
       @NonNull RelayedHttpListenerRequest relayedHttpListenerRequest) {
+    logRequest(relayedHttpListenerRequest, OffsetDateTime.now(), "HTTP_REQUEST");
+    return true;
+  }
+
+  @VisibleForTesting
+  public void logRequest(
+      RelayedHttpListenerRequest relayedHttpListenerRequest,
+      OffsetDateTime requestTimestamp,
+      String prefix) {
+    var headers = relayedHttpListenerRequest.getHeaders();
+    var referer = headers.getOrDefault("Referer", "");
```

Comment:
Are there any case sensitivity issues here?

Reply:
Yeah, good catch, will fix. Annoyingly the headers map is <string, string> so I'll convert to a treemap and a case insensitive compare that way.